src/main.ts

```diff
@@ -6,10 +6,11 @@ async function run(): Promise<void> {
     const token: string = core.getInput('github_token')
     const userName: string = core.getInput('hide_user_name')
     const reason: string = core.getInput('hide_reason')
+    const nbOfCommentsToLeave: number = parseInt(core.getInput('leave_visible'), 10)
 
     const cli = new Client(token)
     const ids = await cli.SelectComments(userName)
-    for (const id of ids) {
+    for (const id of ids.splice(-1, nbOfCommentsToLeave)) {
```

Comment:
```suggestion
    for (const id of ids.splice(-nbOfCommentsToLeave, nbOfCommentsToLeave)) {
```

In order to remove an arbitrary number of elements from the end of the array, I think start also needs to be moved back a `nbOfCommentsToLeave`.

Reply:
You are absolutely correct!